File: viewpager2/src/main/java/com/example/viewpager2/MainActivity.java
package com.example.viewpager2;

import androidx.appcompat.app.AppCompatActivity;
import androidx.viewpager2.widget.ViewPager2;

import android.os.Bundle;

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        ViewPager2 viewPager2=findViewById(R.id.viewpager);
        ViewPager2Adapter viewPager2Adapter=new ViewPager2Adapter();//创建一个设配器
        viewPager2.setAdapter(viewPager2Adapter);
    }
}

File: application2/src/main/java/com/example/application2/MainActivity.kt
package com.example.application2

import android.content.BroadcastReceiver
import android.content.Context
import android.content.Intent
import android.content.IntentFilter
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.util.Log

class MainActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        lateinit var intentFilter:IntentFilter
        intentFilter.addAction("android.test")
        registerReceiver(TestBroadcastReceiver(),intentFilter)
    }

    override fun onDestroy() {
        super.onDestroy()
        if(TestBroadcastReceiver()!=null){
            unregisterReceiver(TestBroadcastReceiver())
        }
    }


    private class TestBroadcastReceiver:BroadcastReceiver(){
        override fun onReceive(context: Context?, intent: Intent?) {
            TODO("Not yet implemented")
            intent!!.action?.let { Log.d("test", it) }
        }

    }


}